src/main/java/blf/core/writers/DataWriter.java

```diff
@@ -43,10 +43,8 @@ public final void endBlock() throws Throwable {
     }
 
     public final void endProgram() throws Throwable {
-        if (this.emissionMode == EmissionSettings.EmissionMode.STREAMING) {
```

Comment:
Oh my, thanks, I didn't see that for some reason. It should have been !=. But I guess it also works to always write the files at the end

Reply:
Since you always delete the Writer state in streaming mode, the writeAll() in endProgram() does not do anything anyway ^^